xds/src/main/java/io/grpc/xds/internal/rbac/engine/GrpcAuthorizationEngine.java

```diff
@@ -335,15 +335,17 @@ private String getPath() {
 
     @Nullable
     private String getHeader(String headerName) {
-      if ("TE".equals(headerName)) {
+      headerName = headerName.toLowerCase();
+      if ("te".equals(headerName)) {
         return null;
       }
-      if ("Host".equals(headerName)) {
-        headerName = "authority";
+      if ("host".equals(headerName)) {
+        return serverCall.getAuthority();
       }
       String value = deserializeHeader(headerName);
       if (value == null && ":method".equals(headerName)) {
-        return "POST";
+        String apiMethod = serverCall.getMethodDescriptor().getFullMethodName();
```

Comment:
No, no, no. This is the HTTP method, like GET/PUT/POST. It has no relation to the grpc method. This was correct before.

Reply:
lol, okay, I keep misunderstand it. It might help if no `:method` in the first sentence. How about ":path" just added.
```
For this design, headers can include :method, :authority, and :path matchers and they should match the values received on-the-wire independent of whether they are stored in Metadata or in separate APIs. :method can be hard-coded to POST if unavailable and a code audit confirms the server denies requests for all other method types. 
```
.

Reply:
What do you mean, "no `:method` in the first sentence"? It absolutely needs to be there, otherwise you wouldn't have this code here.

Seems you are missing `:authority` still as well.

Seems you should remove the `value == null` part for `:method` and move next to `:path`, now that I can compare to your intentions for `:path`. `:method` will always be on-the-wire. The "if unavailable" was "if you don't know the value", and it will never be returned from `deserializeHeader()` as creating the Key will always throw IllegalArgumentException.

Reply:
Right.
I meant that `headers can include :method....independent of whether they are stored in Metadata or in separate APIs` makes me think that if header matcher has a name `:method` then they are stored in API: `serverCall.getMethodDescriptor().getFullMethodName()`, similar to `:authority` -> `serverCall.getAuthority()`, or `:path`. But that was wrong.

Reply:
Remember, this is `:path`:
https://github.com/grpc/grpc-java/blob/fcf13952bb2e2e5e18c20e71735ed0ffd894a2e1/xds/src/main/java/io/grpc/xds/internal/rbac/engine/GrpcAuthorizationEngine.java#L262-L264

If you saw that inline instead of "hidden" behind the getPath() method, it would have probably been more obvious that something was wrong.